### ISSUE TITLE
Implementing the POST APIs for skin changes.

### ISSUE THREAD

**Issue by dscalzi** [astra137/node-mojang#7]
Hello guys,

I've been working on implementing new wrappers to easily handle skin changes for users of this package. I have successfully wrote the methods for Change Skin and Upload Skin although I get the same error response when I send requests to them.

    {"error":"Forbidden","errorMessage":"Current IP not secured"}

I have tried using the curl example provided on http://wiki.vg/Mojang_API#Change_Skin although I get the same issue. The only thing I can think of is that mojang is blocking access to this request. They shouldn't be doing this, though, because on the wiki page there is nothing specifiying that this api is not public. I've found no online resources except for an [issue on their JIRA](https://bugs.mojang.com/browse/WEB-726) which describes the exact problem I am encountering. They marked it resolved with "works as intended", which makes me wonder if this is actually not a public api request.

Also this http://wiki.vg/Talk:Mojang_API
Deobfuscated code from the above link: https://hastebin.com/pawoyagewu.js

This is my implementation:

```javascript
'use strict';

const request = require('./_request');

/**
 * 
 */
function changeskin(uuid, accessToken, slim, url){
    return request({
        method: 'POST',
        hostname: 'api.mojang.com',
        path: '/user/profile/' + uuid + '/skin?model=' + (slim ? 'slim' : '') + '&url=' +encodeURIComponent(url),
        headers: {
            Authorization: ' Bearer ' + accessToken,
        }
    });
}

module.exports = changeskin;```

**Comment by modev** [comment]
I don't know if this really causes your error, but maybe it helps to take a look at [minecraft.net](https://minecraft.netl). When you inspect the requests triggered by a skin change, you can see that they are using  `Content-Type: multipart/form-data` rather than `Content-Type: application/json` (the default header of [_request.js](https://github.com/jamen/node-mojang/blob/master/lib/_request.js#L9))

Since you are passing the form parameters encoded in the URL, you should try using `Content-Type: application/x-www-form-urlencoded`

**Comment by dscalzi** [comment]
Same issue

**Comment by astra137** [comment]
`GET https://api.mojang.com/user/security/location` => is IP secure
`GET https://api.mojang.com/user/security/challenges` => question list
`POST https://api.mojang.com/user/security/location` => submit your answers

I'm including functions for performing these steps in my fork. Still working on it, but I'm making headway. Hopefully it won't all be so drastic that I can't request a pull, but we'll see.

**Comment by dscalzi** [comment]
Resolved by #13 thanks to @maccelerated :)

**Comment by astra137** [comment]
Setting a skin to a URL is implemented, but uploading a local file as a skin isn't. I think it should be added. I want to make a branch for the feat and implement it. Any objections?

**Comment by dscalzi** [comment]
Not at all, I'll reopen this while that is pending. Also, send me a message with your npm username and I can add you to the project. That way you have access to publish releases and such.

**Comment by astra137** [comment]
@dscalzi I am [~maccelerated](https://www.npmjs.com/~maccelerated) on npm.

**Comment by dscalzi** [comment]
All set, I've also released a small update (2.0.1) to reflect the repository transfer 👍
